include/ada/url_pattern_regex.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * @file url_search_params.h
+ * @brief Declaration for the URL Search Params
+ */
+#ifndef ADA_URL_PATTERN_REGEX_H
+#define ADA_URL_PATTERN_REGEX_H
+
+#include <regex>
+
+namespace ada::url_pattern_regex {
+
+class provider {
+  struct type {};
+
+  std::optional<type> create_regex_instance(std::string_view pattern,
+                                            bool ignore_case);
+
+  std::optional<std::vector<std::string>> regex_search(std::string_view input, std::string_view pattern);
```

Comment:
Overall this looks fine. I'm not entirely in love with the `std::optional<std::vector<...>>` part but can live with it.

Reply:
I agree with @jasnell ... I can't imagine that the return type is ideal. But I take it that it is only a prototype.